### PR TITLE
Replace some of the constants with static types

### PR DESCRIPTION
**Pull request by Sheeo**
Makes for better type inference, since `const` declarations are inferred
to their narrowest type at declaration.

E.g. `const x = 5;` infers the type of `x` to be `5`, making the literal
`5` compatible.

Since types are generally widened without a problem, this shouldn't
be an incompatible change with anyones codebase -- although number type
literals are a relatively new typescript feature, so full utilization of
this may require TS 2.1.

### CODE REVIEW

File: src/constants.ts
Comment (Dessix) [inline src/constants.ts]: Can we match the style of `RAMPART_HITS_MAX` here?

File: src/constants.ts
Comment (Dessix) [inline src/constants.ts]: I'll see about adding this one in with my issue-#91 changes, which should present a unified type for many of these items.

File: dist/screeps.d.ts
Comment (Dessix) [inline dist/screeps.d.ts:44]: ~Did we not have this one originally?~
Nice catch, we didn't have this. Value is correct.

**Review by Dessix — CHANGES_REQUESTED** [review]
Overall, I like it! I've noted a few minor stylistic changes, and some observations - We were missing some things. This is a big change, so I'd like multiple contributors' feedback on this (@anisoptera).

File: dist/screeps.d.ts
Comment (Sheeo) [reply dist/screeps.d.ts:44]: Indeed, this was missing.

File: src/constants.ts
Comment (Sheeo) [reply src/constants.ts]: Of course, done.

**Review by Dessix — APPROVED** [review]
Approved. Still need an approval from @anisoptera, @ags131, or @NhanHo before merge due to change scope.

**Comment by anisoptera** [comment]
My one concern is what happens if my codebase runs in an environment with different universal constants. Does this affect the final code generation in any way?

If not, then I'm totally down for this. Thanks for the metric ton of work!

**Comment by Dessix** [comment]
Type literals only affect the validity of operations on that type- from what I can tell, typescript doesn't consider multiple values of the same type as different types (Take a number and compare it against 1, the result isn't the interface of number differenced by the interface of 1 if it's false, or the interface of 1 if it's true). It should be fine because the types are elided, the only situation that would change here is when you change the values acceptable as members of unions of values, or if the value is limited on assignment. (You can't assign `3` to `1 | 2 | 4`, and if you change the type-literal of `MOVE` to ``Derp`` in your universe, `"move"` will no longer be a valid creep body part string literal, and `"Derp"` would be typed directly as `MOVE`). Inheritance/membership chains could also fail on types (see with MOVE- the reason for the issue is that we would be registering constants for each bodypart under a shared type which is simply a union of their individual types. This could be compensated for on a per-custom-environment basis with a second ambient declaration of the interface(s) or type(s) affected.

**Comment by anisoptera** [comment]
Sounds about right, just wanted to make sure. 👍

**Pull request merged**
